### PR TITLE
Allow additionalSp property in advancedConfigurations

### PR DESCRIPTION
**Pull request by VivekVinushanth**

### CODE REVIEW

File: apps/console/src/features/applications/components/edit-application.tsx [hunk 1]
```diff
@@ -374,6 +375,18 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
...
+
+        if (application?.advancedConfigurations?.additionalSpProperties?.length > 0) {
+            application?.advancedConfigurations?.additionalSpProperties?.
+            forEach((item, index) => {
+                if (item.name === "isEnterpriseLoginManagementApp" && item.value === "true") {
+                    setIsEnterpriseLoginMgt(true);
+                }
+            });
+        }
+    }, [application]);
```
Comment (brionmario) [inline apps/console/src/features/applications/components/edit-application.tsx:388]: There should be a lint warning here. Check if ESLint is installed in the IDE.
```suggestion
    }, [ application ]);
```

File: apps/console/src/features/applications/components/forms/general-details-form.tsx [hunk 2]
```diff
@@ -360,8 +365,9 @@ export const GeneralDetailsForm: FunctionComponent<GeneralDetailsFormPopsInterfa
                         ".placeholder")
                 }
                 value={ accessUrl }
-                readOnly={ !hasRequiredScope || ( readOnly && applicationConfig.generalSettings.getFieldReadOnlyStatus(
-                     name, "ACCESS_URL"))}
+                readOnly={ !hasRequiredScope || !isEnterpriseLoginMgtApp ||
+                    ( readOnly && applicationConfig.generalSettings.getFieldReadOnlyStatus(
```
Comment (brionmario) [inline apps/console/src/features/applications/components/forms/general-details-form.tsx:369]:
```suggestion
                    (readOnly && applicationConfig.generalSettings.getFieldReadOnlyStatus(
```

File: apps/console/src/features/applications/models/application.ts [hunk 3]
```diff
@@ -206,6 +207,15 @@ export interface AdvancedConfigurationsInterface {
     skipLogoutConsent?: boolean;
     returnAuthenticatedIdpList?: boolean;
     enableAuthorization?: boolean;
+    additionalSpProperties?: additionalSpProperty[]
+}
+/**
+ * Interface for the additional sp properties.
+ */
+export interface additionalSpProperty {
```
Comment (thivi) [inline apps/console/src/features/applications/models/application.ts:215]:
```suggestion
export interface AdditionalSpProperty {
```

**Comment by malshan1998** [comment]
Tested and working fine.

**Comment by VivekVinushanth** [comment]
Closing this PR since the PR had messedup and unable to recover. Opened a new one 
- https://github.com/wso2/identity-apps/pull/3285

**Pull request closed without merging**